src/coins.cpp

```diff
@@ -659,6 +665,27 @@ bool CCoinsViewCache::GetSidechain(const uint256 & scId, CSidechain& targetScInf
     return false;
 }
 
+int64_t CCoinsViewCache::GetTopQualityCert(const uint256& scId, int epochNumber, uint256& hash) const
+{
+    LogPrint("mempool", "%s.%s():%d - sc %s, epoch %d\n", __FILE__, __func__, __LINE__, scId.ToString(), epochNumber);
+
+    int64_t topQual = base->GetTopQualityCert(scId, epochNumber, hash);
```

Comment:
Since GetTopQualityCert is not implemented for CCoinsViewDB, this call is essentially an expensive constructor of type int64_t topQual = CCoinsView::GetTopQualityCert, i.e. int64_t topQual = CScCertificate::QUALITY_NULL;
Just drop this, it's useless, I believe